cedar-policy-validator/src/custom_schema/fmt.rs

```diff
@@ -0,0 +1,198 @@
+use std::{collections::HashSet, fmt::Display};
+
+use nonempty::NonEmpty;
+use smol_str::SmolStr;
+use thiserror::Error;
+
+use crate::{
+    ActionType, EntityType, NamespaceDefinition, SchemaError, SchemaFragment, SchemaType,
+    SchemaTypeVariant, ValidatorSchema,
+};
+
+impl Display for SchemaFragment {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        for (ns, def) in &self.0 {
+            if ns.is_empty() {
+                write!(f, "{def}")?
+            } else {
+                write!(f, "namespace {ns} {{{def}}}")?
+            }
+        }
+        Ok(())
+    }
+}
+
+impl Display for NamespaceDefinition {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        for (n, ty) in &self.common_types {
+            writeln!(f, "type {n} = {ty};")?
+        }
+        for (n, ty) in &self.entity_types {
+            writeln!(f, "entity {n} {ty};")?
+        }
+        for (n, a) in &self.actions {
+            writeln!(f, "action \"{}\" {a};", n.escape_debug())?
+        }
+        Ok(())
+    }
+}
+
+impl Display for SchemaType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            SchemaType::Type(ty) => match ty {
+                SchemaTypeVariant::Boolean => write!(f, "__cedar::Bool"),
+                SchemaTypeVariant::Entity { name } => write!(f, "{name}"),
+                SchemaTypeVariant::Extension { name } => write!(f, "__cedar::{name}"),
+                SchemaTypeVariant::Long => write!(f, "__cedar::Long"),
```

Comment:
it would be great if we didn't always add `__cedar`, and only add it when necessary.  99.9% of customers will not need `__cedar` and it is ugly.